app/controllers/products_controller.rb

```diff
@@ -1,13 +1,15 @@
 class ProductsController < ApplicationController
   def index
-    @products = Product.all
-    @ladies_products = Product.where(category: 'ladies') # Adjust the condition as per your schema
-    @gents_products = Product.where(category: 'gents')
+    @q = Product.ransack(params[:q])
+    @products = @q.result(distinct: true) # This line should handle the search
+
+    @ladies_products = @products.where(category: 'ladies') # Filter the searched results for ladies
```

Comment:
What is the purpose of these 2 variables, are we showing both on FE or one at a time?

Reply:
Which 2 variables are you talking about? @q and @products are using for ransack.